### ISSUE TITLE
Wrong image resolution using the image tool

### ISSUE THREAD

**Issue by cmaziero** [xournalpp/xournalpp#3022]
**Affects versions :**
 - OS: Linux Ubuntu 20.04 64 bit (not tested in other systems)
 - Desktop environment: Gnome-Wayland (default in Ubuntu 20.04?)
 - Which version of libgtk do you use: 3.24.20
 - Version of Xournal++: 1.0.20
 - Installation method: DEB package manually installed from PPA repo

**Describe the bug**
When inserting images from the same PNG file in a PDF document using the "image tool" of Xournalpp, each insertion gets a different size: some are smaller and others bigger, although the PNG file is the same.

**To reproduce**
1. open a PDF document in Xournalpp
2. select the "image tool"
3. click on the document, it will open a file dialog
4. in the file dialog select a PNG image to insert
5. click "open" to insert the image
6. repeat steps 3-5 to insert the **same image** again
7. the two insertions of the same PNG image do not have the same dimension
**Expected behavior**
All insertions of the same image (i.e. same file) in a document should have the same dimensions.

**Screenshots of Problem**
The screenshot below shows 3 insertions of the same image on a PDF document. All three insertions have distinct dimensions; the expected behavior would be to have all insertions displayed with the same dimensions.

![screenshot](https://user-images.githubusercontent.com/33373746/114215551-1b4c3a00-993c-11eb-9f38-38112401f707.png)

**Additional context**
Add any other information or context about the problem here.

**Comment by mjg** [comment]
xournalpp computes the scale as follows: if the image fits into the page as is, it is not scaled (`zoom==1`). If not then it is scaled so that it fits. Two caveats:

- 1 image pixel is one "page point", that is xournalpp does not read any dpi from the image but uses it's own dpi value.
- "fits into the page" means "at the point of insertion".

So, if you insert a "large" image at different positions you get a different scale. This is definitely not a bug but exactly what the code intentionally does.

I'm wondering whether it's really useful, though: You can always rescale a large image from the top left corner (the insertion point), so the current behaviour is not strictly necessary to allow you a rescaling. It makes the rescaling easier for large images (or those inserted near the right or bottom edge) but basically forces you watch your point of insertion...

OTOH, you can always scale once and then copy the scaled image instead of re-inserting it. That's the quickest fix for your use-case.

**Comment by cmaziero** [comment]
Hi mjg, thanks for the reply.

I use xournalpp mostly to make annotations in PDF files, like paper reviews and students' manuscripts. Also, as a coordinator of an undergraduate course, I frequently use it to add my signature and stamps in official documents (it would be better to use digital signatures, but our administration is that it is...). Most of such documents are A4 paper-sized.

The autoscaling feature of xournalpp makes it very annoying to add signatures or stamps, because every time I add one, it's size changes unpredictably. Also, to have sharper images, my signature/stamp image files are PNG with transparency and higher resolution (800 pixels-wide or higher), producing big images that should be manually rescaled every time.

To solve my issue temporarily, I modified the older xournal source code to use a fixed zoom factor (I simply modified line 59 of `xo-image.c` to "`scale = 0.25 ;`". It is not ideal, but it's working.

IMHO, a preferrable solution would be xournalpp to have a "zoom factor" preference for the image tool, allowing to choose between "auto" and some user-defined numeric value.

Best regards!

**Comment by mjg** [comment]
Absolutely, @cmaziero This repo is turning into a lecturers' meeting place, we all have similar use cases :)

I checked out this bug to see whether it is solved in the development version, and used the source to confirm it behaves as intended. I think we could use 2 things in the preference:
- default zoom factor (currently "1", meaning 72 dpi)
- auto scale to fit page (currently on)

So, this is rather an RFE than a bug report.

**Comment by cmaziero** [comment]
Michael, you're right. If it does what it is intended to do, it's not a bug, it's a feature! :smile:

Should I change this issue's label from "bug" to "enhancement", or just close it and create another one?

Btw, How to I reclassify an issue in GH?

Thanks again!

**Comment by mjg** [comment]
I'm not sure we "ordinary people" can change the label. I'd suggest you edit the title to reflect the feature request. This way everything is still together here - it's relevant to the RFE, and the project maintainers can change the label when they see it. (I think they're preparing for a new release, so squashing bugs and postponing RFEs.)

**Comment by cmaziero** [comment]
Done. I'll keep using the oldtimer xournal while waiting for this to be taken into account (no time to code by now).

Regards!

**Comment by madduck** [comment]
My [comment in #4622](https://github.com/xournalpp/xournalpp/issues/4622#issuecomment-2244660011) makes the case for having stamps/images replace the mouse pointer on insert, and allowing quick scaling with the scroll wheel, before the stamp/image is placed with a mouse click.
